Documentation/gettingstarted/bgp-control-plane.rst

```diff
@@ -0,0 +1,330 @@
+
+When these CRDs are written or read from the cluster the ``Controllers``
+will take notice and perform the necessary actions to drive the
+``BGP Control Plane`` to the desired state described by the policy.
+
+The policy in ``yaml`` form is defined below:
+
+.. code-block:: yaml
+
+   apiVersion: "cilium.io/v2alpha1"
+   kind: CiliumBGPPeeringPolicy
+   metadata:
+    name: 01-bgp-peering-policy
+   spec: # CiliumBGPPeeringPolicySpec
+    nodeSelector:
+      matchLabels:
+        bgp-policy: a
+    virtualRouters: # []CiliumBGPVirtualRouter
+    - localASN: 64512
+      exportPodCIDR: true
+      neighbors: # []CiliumBGPNeighbor
+       - peerAddress: 'fc00:f853:ccd:e793::50/128'
+         peerASN: 64512
+
+Fields
+^^^^^^
+
+::
+
+   nodeSelector: Nodes which are selected by this label selector will apply the given policy
+    
+    virtualRouters: One or more peering configurations outlined below. Each peering configuration can be thought of as a BGP router instance.
+       
+       virtualRouters[*].localASN: The local ASN for this peering configuration
+       
+       virtualRouters[*].exportPodCIDR: Whether to export the private pod CIDR block to the listed neighbors
+       
+       virtualRouters[*].neighbors: A list of neighbors to peer with
+           neighbors[*].peerAddress: The address of the peer neighbor
+           neighbors[*].peerASN: The ASN of the peer   
+
+.. note:: 
+
+   Setting unique configuration details of a particular
+   instantiated virtual router on a particular Cilium node is explained
+   in `Virtual Router Attributes <#Virtual%20Router%20Attributes>`__
+
+Creating a BGP Topology
+-----------------------
+
+Rules
+~~~~~
+
+Follow the rules below to have a ``CiliumBGPPeeringPolicy`` correctly
+apply to a node.
+
+-  Only a single ``CiliumBGPPeeringPolicy`` can apply to a ``Cilium``
+   node.
+
+   -  If the ``BGP Control Plane`` on a node iterates through the
+      ``CiliumBGPPeeringPolicy`` CRs currently written to the cluster
+      and discovers (n > 1) policies match its labels, it will return an
+      error and remove any existing BGP sessions. Only (n == 1) policies
+      **must** match a node's label sets.
+   -  Administrators should test a new BGP topology in a staging
+      environment before making permanent changes in production.
+
+-  Within a ``CiliumBGPPeeringPolicy`` each ``CiliumBGPVirtualRouter``
+   defined must have a unique ``localASN`` field.
+
+   -  A node cannot host two or more logical routers with the same local
+      ASN. Local ASNs are used as unique keys for a logical router.
+   -  A node can define the remote ASN on a per-neighbor basis to
+      mitigate this scenario. See ``CiliumBGPNeighbor`` CR
+      sub-structure.
+
+-  IPv6 single stack deployments **must** set an IPv4 encoded
+   ``routerID`` field in each defined ``CiliumBGPVirtualRouter`` object
+   within a ``CiliumBGPPeeringPolicy``
+
+   -  Cilium running on a IPv6 single stack cluster cannot reliably
```

Comment:
Calico uses the hash of the ipv6 address to derive a router-id if it's not provided.   Defining a router-id for each server is rather annoying especially in presence of auto-scaling and ephemeral nodes.  Would it be possible to have some form of "IPAM" for router-IDs?   Having to manually set them does not scale well :(

Reply:
This is a good point. 

We have discussed a bit about IPAM for router-IDs, and it definitely wasn't a no-go. More, a time constraint desicion not to add it for this initial merge. 

The hashing of the IPv6 address is an interesting one for an "auto generated" way of going about it. 

I think we had discussed with our teams how network admins may *want* to hard code router ids, as they pen-and-paper design their networks. Generation can causes surprises in some cases. 

I'd be happy to cater to both use cases, ones where the net admins want to be completely explicit, and ones where generation is OK and helps lessen the administrative burden.